5.machine_learning/1.evaluate_models/scripts/3.ROC_QC_new_method.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Perform bootstrapping ROC AUC method to determine if QC significantly improves performance of model
+# 
+# In this method, we have trained two models; one with QC'd data and the other without QC'd data (more noise). 
+# To assess quality, apply the models to a holdout dataset that both models have never seen that has been QC'd.
+# We take subsamples of this dataset to perform a bootstapping method that will calculate the ROC AUC across the QC'd holdout samples.
+# 
+# The goal of this analysis is to see how well the models perform when applied to a new dataset that has only "good" cells, since we will be applying QC to all future plates.
+# 
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+import joblib
+from joblib import load
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from sklearn.metrics import roc_auc_score
+from sklearn.utils import resample
+import matplotlib.colors as mcolors
+from scipy.stats import ttest_ind
+
+import sys
+
+sys.path.append("../../utils")
+from training_utils import get_X_y_data
+
+
+# ## Define function
+
+# In[2]:
+
+
+# Define function to preform bootstrapping
+def bootstrap_roc_auc(y_true, y_pred, n_bootstraps=1000):
```

Comment:
probably best to not define this twice but import it instead

Reply:
Agreed, adding utility now!